### PR TITLE
Use hash code operator for hashing array elements

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
@cberner can you please take a look?

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayHashCodeOperator.java [hunk 1]
```diff
@@ -45,16 +48,16 @@ private ArrayHashCodeOperator()
...
-    public static long hash(Type type, Block block)
+    public static long hash(MethodHandle hashCodeFunction, Type type, Block block)
     {
-        int hash = 0;
+        long hash = 0;
         for (int i = 0; i < block.getPositionCount(); i++) {
             checkElementNotNull(block.isNull(i), ARRAY_NULL_ELEMENT_MSG);
-            // TODO: This should be migrated away from Type.hash and invoke a MethodHandle for the hash code operator instead
-            hash = (int) CombineHashFunction.getHash(hash, type.hash(block, i));
+            hash += hashPosition(hashCodeFunction, type, block, i);
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayHashCodeOperator.java]: I just summed the hashes similar to `MapHashCodeOperator` does. Let me know if we want to use `CombineHashFunction` instead.

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayHashCodeOperator.java]: It needs to behave exactly the same as `ArrayType.hash`, so ya we should keep using `CombineHashFunction` and an `int`

**Comment by nezihyigitbasi** [comment]
@cberner updated.

**Comment by cberner** [comment]
Merged. Thanks!

**Pull request closed without merging**